src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -520,6 +520,48 @@ private void Mvlocfld(LocalBuilder lt, FieldInfo ft)
 
         private LocalBuilder DeclareReadOnlySpanChar() => _ilg!.DeclareLocal(typeof(ReadOnlySpan<char>));
 
+        /// <summary>Rents an Int32 local variable slot from the pool of locals.</summary>
+        /// <remarks>
+        /// Care must be taken to Dispose of the returned <see cref="RentedLocalBuilder"/> when it's no longer needed,
+        /// and also not to jump into the middle of a block involving a rented local from outside of that block.
+        /// </remarks>
+        private RentedLocalBuilder RentInt32Local() => new RentedLocalBuilder(
+            _int32LocalsPool ??= new Stack<LocalBuilder>(),
+            _int32LocalsPool.TryPop(out LocalBuilder? iterationLocal) ? iterationLocal : DeclareInt32());
+
+        /// <summary>Rents an ReadOnlySpan(char) local variable slot from the pool of locals.</summary>
+        /// <remarks>
+        /// Care must be taken to Dispose of the returned <see cref="RentedLocalBuilder"/> when it's no longer needed,
+        /// and also not to jump into the middle of a block involving a rented local from outside of that block.
+        /// </remarks>
+        private RentedLocalBuilder RentReadOnlySpanCharLocal() => new RentedLocalBuilder(
+            _readOnlySpanCharLocalsPool ??= new Stack<LocalBuilder>(1),
```

Comment:
Why `(1)` here, but default above -- because you expect to need only 1-ish of these, vs 4-ish?

Reply:
Correction, I see that the default constructor does not allocate an array at all.

Reply:
> Why (1) here, but default above -- because you expect to need only 1-ish of these, vs 4-ish?

Yup.  We typically create 3 or 4 Int32 locals.  We generally don't need more than 1 span.

Reply:
Might be worth a note.